### PR TITLE
Add eigfact in a dumb way

### PR DESCRIPTION
**Pull request by kshyatt**
I got mad that we don't have this, but it's pretty simple to do for the single-matrix case. Hope it's ok.

### CODE REVIEW

File: test/eigen.jl [hunk 1]
```diff
@@ -5,10 +5,16 @@
         @test vals === SVector(2.0)
         @test eigvals(m) === vals
         @test vecs === SMatrix{1,1}(1.0)
+        ef = eigfact(m)
+        @test ef[:values] === SVector(2.0)
+        @test ef[:vecs] === SMatrix{1,1}(1.0)
```
Comment (andyferris) [inline test/eigen.jl]: Tests say `:vecs` doesn't exist as a key

Comment (fredrikekre) [inline test/eigen.jl]: should be `:vectors` :)

**Comment by kshyatt** [comment]
There's another problem with the symmetric tests I'm trying to fix, stand
by for updates :)

**Comment by kshyatt** [comment]
CI passed! Good to go?

**Pull request merged**